src/Storages/MergeTree/MergeTreeDataSelectExecutor.cpp

```diff
@@ -746,8 +747,17 @@ RangesInDataParts MergeTreeDataSelectExecutor::filterPartsByPrimaryKeyAndSkipInd
                 CurrentMetrics::MergeTreeDataSelectExecutorThreadsScheduled,
                 num_threads);
 
+            SCOPE_EXIT({ pool.wait(); });
```

Comment:
Here is an interesting case could be.

`pool.wait();` -- can throw an exception from other failed task. You dont what it when you already got some from `pool.scheduleOrThrow()`.

Just leave that as it was. Than d-tor ThreadPool will wait and mute other exception.